Bottom: 8b7f20545a233ec930ffd0b00fae0fe6bd660a5e
Top:    b3d5a2afe071cd15b927b29ec845455143ca14b0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-26 23:21:03 +0100

Refresh of patch

---

diff --git a/src/const.h b/src/const.h
index 76db89a..6fa7cbc 100644
--- a/src/const.h
+++ b/src/const.h
@@ -39,8 +39,10 @@
  * the layer-numbers of the two additional special layers
  * 'top' and 'bottom'. The offset of MAX_LAYER is not added
  */
-#define	BOTTOM_LAYER		0
-#define	TOP_LAYER		1
+#define	BOTTOM_SILK_LAYER	0
+#define	TOP_SILK_LAYER		1
+#define	BOTTOM_SOLDERMASK_LAYER	2
+#define	TOP_SOLDERMASK_LAYER	3
 
 /* ---------------------------------------------------------------------------
  * misc constants
diff --git a/src/data.h b/src/data.h
index 470933b..89d9b87 100644
--- a/src/data.h
+++ b/src/data.h
@@ -47,8 +47,8 @@ extern PCBType *PCB;
 
 #define max_group (PCB->Data->LayerN)
 #define max_copper_layer (PCB->Data->LayerN)
-#define bottom_silk_layer (max_copper_layer + BOTTOM_LAYER)
-#define top_silk_layer (max_copper_layer + TOP_LAYER)
+#define bottom_silk_layer (max_copper_layer + BOTTOM_SILK_LAYER)
+#define top_silk_layer (max_copper_layer + TOP_SILK_LAYER)
 #define bottom_soldermask_layer (max_copper_layer + BOTTOM_SOLDERMASK_LAYER)
 #define top_soldermask_layer (max_copper_layer + TOP_SOLDERMASK_LAYER)
 
diff --git a/src/find.c b/src/find.c
index f11f3d6..f797cc9 100644
--- a/src/find.c
+++ b/src/find.c
@@ -124,6 +124,9 @@
 #define	IS_PV_ON_PAD(PV,Pad) \
 	( IsPointInPad((PV)->X, (PV)->Y, MAX((PV)->Thickness/2 +Bloat,0), (Pad)))
 
+#define BOTTOM_LAYER 0
+#define TOP_LAYER 1
+
 static DrcViolationType
 *pcb_drc_violation_new (const char *title,
                         const char *explanation,
diff --git a/src/fontmode.c b/src/fontmode.c
index 4eabb9c..4afd3cd 100644
--- a/src/fontmode.c
+++ b/src/fontmode.c
@@ -88,8 +88,8 @@ FontEdit (int argc, char **argv, Coord Ux, Coord Uy)
   Settings.minWid = PCB->minWid = 1;
   Settings.minSlk = PCB->minSlk = 1;
 
-  MoveLayerToGroup (max_copper_layer + TOP_LAYER, 0);
-  MoveLayerToGroup (max_copper_layer + BOTTOM_LAYER, 1);
+  MoveLayerToGroup (max_copper_layer + TOP_SILK_LAYER, 0);
+  MoveLayerToGroup (max_copper_layer + BOTTOM_SILK_LAYER, 1);
 
   while (PCB->Data->LayerN > 4)
     MoveLayer (4, -1);
diff --git a/src/macro.h b/src/macro.h
index 1b352e3..d5c57e9 100644
--- a/src/macro.h
+++ b/src/macro.h
@@ -152,11 +152,11 @@ extern int mem_any_set (unsigned char *, int);
 	((TEST_FLAG(ONSOLDERFLAG, (o)) != 0) == SWAP_IDENT)
 
 /* ---------------------------------------------------------------------------
- *  Determines if an object is on the given side. side is either BOTTOM_LAYER
- *  or TOP_LAYER.
+ *  Determines if an object is on the given side. side is either BOTTOM_GROUP
+ *  or TOP_GROUP.
  */
 #define ON_SIDE(element, side) \
-        (TEST_FLAG (ONSOLDERFLAG, element) == (side == BOTTOM_LAYER))
+        (TEST_FLAG (ONSOLDERFLAG, element) == (side == BOTTOM_GROUP))
 
 /* ---------------------------------------------------------------------------
  * some loop shortcuts
diff --git a/src/misc.c b/src/misc.c
index b23a6f2..a2e55f2 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1019,7 +1019,7 @@ ParseGroupString (char *s, LayerGroupType *LayerGroup, int LayerN)
             case 'C':
             case 't':
             case 'T':
-              layer = LayerN + TOP_LAYER;
+              layer = LayerN + TOP_SILK_LAYER;
               c_set = true;
               break;
 
@@ -1027,7 +1027,7 @@ ParseGroupString (char *s, LayerGroupType *LayerGroup, int LayerN)
             case 'S':
             case 'b':
             case 'B':
-              layer = LayerN + BOTTOM_LAYER;
+              layer = LayerN + BOTTOM_SILK_LAYER;
               s_set = true;
               break;
 
@@ -1037,7 +1037,7 @@ ParseGroupString (char *s, LayerGroupType *LayerGroup, int LayerN)
               layer = atoi (s) - 1;
               break;
             }
-          if (layer > LayerN + MAX (BOTTOM_LAYER, TOP_LAYER) ||
+          if (layer >= LayerN + 2 ||
               member >= LayerN + 1)
             goto error;
           groupnum[layer] = group;
@@ -1058,11 +1058,11 @@ ParseGroupString (char *s, LayerGroupType *LayerGroup, int LayerN)
     }
   if (!s_set)
     LayerGroup->Entries[BOTTOM_LAYER][LayerGroup->Number[BOTTOM_LAYER]++] =
-      LayerN + BOTTOM_LAYER;
+      LayerN + BOTTOM_SILK_LAYER;
   if (!c_set)
     LayerGroup->
       Entries[TOP_LAYER][LayerGroup->Number[TOP_LAYER]++] =
-      LayerN + TOP_LAYER;
+      LayerN + TOP_SILK_LAYER;
 
   for (layer = 0; layer < LayerN && group < LayerN; layer++)
     if (groupnum[layer] == -1)
diff --git a/src/move.c b/src/move.c
index b50531e..ac64063 100644
--- a/src/move.c
+++ b/src/move.c
@@ -910,7 +910,7 @@ move_all_thermals (int old_index, int new_index)
 static int
 LastLayerInTopGroup (int layer)
 {
-  int top_group = GetLayerGroupNumberByNumber(max_group + TOP_LAYER);
+  int top_group = GetLayerGroupNumberByNumber(max_group + TOP_GROUP);
   int lgroup = GetLayerGroupNumberByNumber(layer);
   if (top_group == lgroup
       && PCB->LayerGroups.Number[lgroup] == 2)
@@ -921,7 +921,7 @@ LastLayerInTopGroup (int layer)
 static int
 LastLayerInBottomGroup (int layer)
 {
-  int bottom_group = GetLayerGroupNumberByNumber(max_group + BOTTOM_LAYER);
+  int bottom_group = GetLayerGroupNumberByNumber(max_group + BOTTOM_GROUP);
   int lgroup = GetLayerGroupNumberByNumber(layer);
   if (bottom_group == lgroup
       && PCB->LayerGroups.Number[lgroup] == 2)
diff --git a/src/parse_y.y b/src/parse_y.y
index 6fc0724..72a0046 100644
--- a/src/parse_y.y
+++ b/src/parse_y.y
@@ -1060,7 +1060,7 @@ text_newformat
 				if ($8 & ONSILKFLAG)
 				{
 					LayerType *lay = &yyData->Layer[yyData->LayerN +
-						(($8 & ONSOLDERFLAG) ? BOTTOM_LAYER : TOP_LAYER)];
+						(($8 & ONSOLDERFLAG) ? BOTTOM_SILK_LAYER : TOP_SILK_LAYER)];
 
 					CreateNewText(lay ,yyFont, OU ($3), OU ($4), $5, $6, $7,
 						      OldFlags($8));
@@ -1085,7 +1085,7 @@ text_hi_format
 				if ($8.f & ONSILKFLAG)
 				{
 					LayerType *lay = &yyData->Layer[yyData->LayerN +
-						(($8.f & ONSOLDERFLAG) ? BOTTOM_LAYER : TOP_LAYER)];
+						(($8.f & ONSOLDERFLAG) ? BOTTOM_SILK_LAYER : TOP_SILK_LAYER)];
 
 					CreateNewText(lay, yyFont, NU ($3), NU ($4), $5, $6, $7, $8);
 				}
